activesupport/test/core_ext/module_test.rb

```diff
@@ -440,4 +440,78 @@ def initialize(place)
     assert_not_respond_to place, :the_city
     assert place.respond_to?(:the_city, true)
   end
+
+  def test_private_delegate_with_private_option
+    location = Class.new do
+      def initialize(place)
+        @place = place
+      end
+
+      delegate(:street, :city, to: :@place, private: true)
+    end
+
+    place = location.new(Somewhere.new("Such street", "Sad city"))
+
+    assert_not_respond_to place, :street
+    assert_not_respond_to place, :city
+
+    assert place.respond_to?(:street, true) # Asking for private method
+    assert place.respond_to?(:city, true)
+  end
+
+  def test_some_public_some_private_delegate_with_private_option
+    location = Class.new do
+      def initialize(place)
+        @place = place
+      end
+
+      delegate(:street, to: :@place)
+      delegate(:city, to: :@place, private: true)
+    end
+
+    place = location.new(Somewhere.new("Such street", "Sad city"))
+
+    assert_respond_to place, :street
+    assert_not_respond_to place, :city
+
+    assert place.respond_to?(:city, true) # Asking for private method
+  end
+
+  def test_private_delegate_prefixed_with_private_option
+    location = Class.new do
+      def initialize(place)
+        @place = place
+      end
+
+      delegate(:street, :city, to: :@place, prefix: :the, private: true)
+    end
+
+    place = location.new(Somewhere.new("Such street", "Sad city"))
+
+    assert_not_respond_to place, :street
+    assert_not_respond_to place, :city
+
+    assert_not_respond_to place, :the_street
+    assert place.respond_to?(:the_street, true)
+    assert_not_respond_to place, :the_city
+    assert place.respond_to?(:the_city, true)
+  end
+
+  def test_private_delegate_with_private_option_passed_to_splat_private
+    location = Class.new do
+      def initialize(place)
+        @place = place
+      end
+
+      private(*delegate(:street, :city, to: :@place, private: true))
```

Comment:
What is this testing?

Reply:
as lot of developers were used to writing the private delegated methods this way

```ruby
private *delegate(:city, to: @profile)
```

e.g.: https://stackoverflow.com/a/35858112/473040

I wanted to ensure that nothing will  break when they decide to change it to 

```ruby
private *delegate(:city, to: @profile, private: true)
```

I know that this is enough onece thin PR is there:

```ruby
delegate(:city, to: @profile, private: true)
```

I just wanted to be sure this will not break

```ruby
private *delegate(:city, to: @profile, private: true)
```

I can remove that test if you want, but I've found it valuable (As I've manage to write my first implementation of this PR that was accidentally disabling the splat private format)

Reply:
I don't think we care about `private *delegate(.., private: true)` (double-private), but a test for `private *delegate(..)` seems reasonable.

Reply:
I'll refactor that test to check the return value then

Reply:
ok how about this?: https://github.com/rails/rails/pull/31944/commits/819842708c9bbdfebc34d701017b711fcb232e11